dracut/99emergency-timeout/timeout.sh

```diff
@@ -0,0 +1,90 @@
+# Before starting the emergency shell, prompt the user to press Enter.
+# If they don't, reboot the system.
+#
+# Assumes /bin/sh is bash.
+
+# _wait_for_journalctl_to_stop will block until either:
+# - no messages have appeared in journalctl for the past 5 seconds
+# - 15 seconds have elapsed
+_wait_for_journalctl_to_stop() {
+    local time_since_last_log=0
+
+    local time_started="$(date '+%s')"
+    local now="$(date '+%s')"
+
+    while [ ${time_since_last_log} -lt 5 -a $((now-time_started)) -lt 15 ]; do
+        sleep 1
+
+        local last_log_timestamp="$(journalctl -e -n 1 -q -o short-unix | cut -d '.' -f 1)"
+        local now="$(date '+%s')"
+
+        local time_since_last_log=$((now-last_log_timestamp))
+    done
+}
+
+_prompt_for_timeout() {
+    local timeout=300
+    local interval=15
+
+    if [[ -e /.emergency-shell-confirmed ]]; then
+        return
+    fi
+    ignition_units="ignition-disks.service ignition-files.service ignition-mount.service"
```

Comment:
still needs umount

Reply:
Do we have a separate service for umount? It looks like the `umount` stage is executed within here: https://github.com/coreos/ignition-dracut/blob/master/dracut/30ignition/ignition-mount.service#L29

Reply:
Oh, whoops. My bad.

Reply:
Np, thanks for the review and making me look closer!